### DISCUSSION TITLE
Python Analysis Extra Paths

### DISCUSSION THREAD

**Discussion by karimlevallois** [zed-industries/zed#7848] [Feature Requests]
Hi everyone! Trying out Zed and really liking it so far. Quick question on Python config of I may...

I have a local folder which contains some additional stubs and docstrings for stdlib etc. and would like to bring those into the pyright set-up, just like I have with Neovim and Sublime Text.

This involves adding an additional path pointing to my local folder using 'python.analysis.extraPaths' (https://github.com/microsoft/pyright/blob/main/docs/settings.md)

Does anyone have an idea how to do this?

Many thanks,

Karim

**Comment by karimlevallois** [comment]
Still need help on this if anyone has any ideas?

**Comment by Moshyfawn** [comment]
If I understand you correctly, you should be able to add this configuration to `pyrightconfig.json` so that Zed will pick it up.

**Reply by karimlevallois** [reply]
I’m more after a global setting the same as I can apply to the above-mentioned editors without having to add it to my config files.

**Comment by ariuk44** [comment]
Hello, have you solved this problem, I have the same problem

**Comment by Allerito** [comment]
Hello, I found a way to resolve this issue: simply add this information to the Python LSP. I have only tried it with BasedPyright, but it works.

```json
"lsp": {
    "basedpyright": {
      "settings": {
        "basedpyright.analysis": {
          "extraPaths": [
             // needed paths
          ]
        },
        "basedpyright.autoComplete": {
          "extraPaths": [
             // needed paths
          ]
        }
      }
    }
  }
```

**Reply by thraizz** [reply]
Is this relative from the repository root or absolute?

**Reply by Allerito** [reply]
I used them as absolutes because I needed my internal software libraries.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
